odoo/orm/models.py

```diff
@@ -3787,10 +3793,94 @@ def _filtered_access(self, operation: str):
             self.filtered(lambda record: record.has_access(operation))
 
         """
-        if self and not self.env.su and (result := self._check_access(operation)):
+        if self and not self.env.su and (result := self.__check_access_cached(operation)):
             return self - result[0]
         return self
 
+    def __check_access_cached(self, operation: str) -> tuple[Self, Callable] | None:
+        """ Cached version of `_check_access`. """
+        # get the cache
+        env = self.env
+        if operation == 'read':
+            access = env._access_read[self._name]
+        else:
+            return self._check_access(operation)
+        # note: access[0] is reserved for the model access
+        # - no value => we don't know if we have access to the model
+        # - otherwise => we have access to the model
+
+        # if we have access data for some records, the model is accessible
+        if not access:
+            if result := self.browse()._check_access(operation):
+                return self, result[1]
+            access[0] = False
```

Comment:
If `result` is empty, then the model is accessible.  Why setting `access[0]` to `False`?